### ISSUE TITLE
Slow query during asset manager reindex

### ISSUE THREAD

**Issue by mtneug** [opencast/opencast#4748]
**Describe the bug**

During the asset manager reindex, all snapshots are loaded in memory from the DB. This is done in a single query resulting in a very slow DB query (and possibly in a heap overflow). I think this should be the culprit:

https://github.com/opencast/opencast/blob/cf744c6632e94c47b9adbbb31c3d609986aa01f1/modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/AssetManagerImpl.java#L951

In our CockroachDB cluster, this is reported as this (simplified) query:

```sql
SELECT DISTINCT
       t0.id,
       t0.archival_date,
       t0.availability,
       t0.mediapackage_id,
       t0.mediapackage_xml,
       t0.organization_id,
       t0.owner,
       t0.series_id,
       t0.storage_id,
       t0.version
  FROM oc_assets_snapshot AS t0
 WHERE t0.version
       = (
      SELECT max(t1.version)
        FROM oc_assets_snapshot AS t1
       WHERE t1.mediapackage_id = t0.mediapackage_id
    )
```

**To Reproduce**

1. Ingest videos
2. Perform a reindex

**Expected behavior**

The asset manager reindex uses pagination.

**Server environment:**
- Installation method container
- Version 13.x

**Comment by JamesUoM** [comment]
We're hovering around 300,000 events so this has got to scale!

**Comment by gregorydlogan** [comment]
Fixed with #4783
